apps/total-typescript/src/templates/article-template.tsx

```diff
@@ -193,6 +200,25 @@ const ArticleTemplate: React.FC<ArticleTemplateProps> = ({
                 }}
               />
             )}
+            {!isLoggedIn && (
+              <div className="absolute bottom-0 left-0 z-20 flex h-64 w-full flex-col items-center justify-end bg-gradient-to-t from-background via-background to-transparent p-5 text-center">
+                <div className="rounded-lg border bg-background p-5 shadow-lg">
+                  <h3 className="mb-2 text-lg font-semibold text-white">
+                    Log in to read the full article
+                  </h3>
+                  <p className="mb-4 text-sm text-gray-400">
+                    Unlock this article and more by logging into your account.
```

Comment:
it's an email wall, same as on tutorials and subscribing should unlock the article instantly (instead of redirecting).